### PR TITLE
Automatic extraction for Cityscapes Dataset

### PR DESCRIPTION
**Pull request by RJT1990**
**Status quo :** Requires user to manually extract the Cityscapes folders before use.

**Proposed change :** if the folders are not present, but the Cityscapes zip files are, then these will be extracted to the appropriate folders. This mirrors extraction logic in other datasets - e.g. ImageNet.

### CODE REVIEW

**Comment by soumith** [comment]
thanks!

File: torchvision/datasets/cityscapes.py [hunk 1]
```diff
@@ -127,8 +128,19 @@ def __init__(self, root, split='train', mode='fine', target_type='instance',
                              ' or "color"')
 
         if not os.path.isdir(self.images_dir) or not os.path.isdir(self.targets_dir):
-            raise RuntimeError('Dataset not found or incomplete. Please make sure all required folders for the'
-                               ' specified "split" and "mode" are inside the "root" directory')
+            image_dir_zip = os.path.join(self.root, 'leftImg8bit') + '_trainvaltest.zip'
```
Comment (TheCodez) [inline torchvision/datasets/cityscapes.py]: @RJT1990 this is missing case were we're using the `coarse` set with `train_extra` split. In that case the additional images zip is called `leftImg8bit_trainextra.zip`. Maybe you could add that case as well in a follow up pr?

File: torchvision/datasets/cityscapes.py [hunk 2]
```diff
@@ -191,3 +203,9 @@ def _get_target_suffix(self, mode, target_type):
             return '{}_color.png'.format(mode)
         else:
             return '{}_polygons.json'.format(mode)
+
+
+def extract_cityscapes_zip(zip_location, root):
```
Comment (fmassa) [inline torchvision/datasets/cityscapes.py]: couldn't you use https://github.com/pytorch/vision/blob/867e5b4e7835dbebcf0c501850a7b08e466496f1/torchvision/datasets/utils.py#L215-L236 instead ?

**Comment by RJT1990** [comment]
Thanks @fmassa, @TheCodez - will send a follow-up PR to (a) also cover the coarse/train_extra case and (b) use that existing utility function.

**Pull request merged**